packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -121,12 +121,13 @@ export function TransferPanel() {
   const isSwitchingL2Chain = useIsSwitchingL2Chain()
 
   // Link the amount state directly to the amount in query params -  no need of useState
-  // Both `amount` getter and setter will internally be useing useArbQueryParams functions
+  // Both `amount` getter and setter will internally be using `useArbQueryParams` functions
   const [{ amount }, setQueryParams] = useArbQueryParams()
   const amountNum = parseFloat(amount) // just a numerical variant of amount
   const setAmount = useCallback(
     (newAmount: string) => {
-      setQueryParams({ amount: newAmount })
+      let amount: string = Number(newAmount) >= 0 ? newAmount : ''
```

Comment:
Can we just handle this case in the `useArbQueryParam` decoder itself? Then we wouldn't be needing this extra check anywhere in the app.

Reply:
If we don't put this check here, when the user types -(number) in the input, the URL flashes :(

Reply:
I think @dewanshparashar is probably right and we can do it in the `encode` method and just have it default to `0` if it's invalid?

Reply:
fixed 👍